### DISCUSSION TITLE
adding a code template via Client.java

### DISCUSSION THREAD

**Discussion by mjtrac** [nextgenhealthcare/connect#6015] [General]
Looking at connect/server/src/com/mirth/connect/client/core/Client.java, I see Client calls to retrieve code templates and code template libraries, and calls to update existing code templates and libraries, but no call to add a new code template or code template library.  Is there a way to add a new code template library and code template without using the Administrator to create or import it?

**Comment by jonbartels** [comment]
I know that other parts of Mirth (like SSL certs and channel groups) are retrieved and posted as a complete blob instead of as individual items. Reading the API info:
![image](https://github.com/nextgenhealthcare/connect/assets/7669292/55d620b9-feae-42b5-8c2e-a1568db127d0)

It looks like libraries are fetched as a complete set. Like you're literally updating the whole set of libraries not just one by one.

New code templates can be created by doing a PUT with a generated ID. I don't see a POST, so I think you have to do a PUT and generate your own UUID for the ID.

Each one of these API methods should have a corresponding call in the Client class. HTH

**Comment by mjtrac** [comment]
Thanks, Jon.  Good to know about using a PUT, but do you know whether that
would work to create a new library to hold the template?  I'd like to have a channel create its own
code template library with a code template, without potentially affecting any existing code templates in other code template libraries. I guess I'll give it a try, but something like 'createCodeTemplateLibrary' and 'createCodeTemplateInLibrary' would be nice.
